src/Mod/Assembly/JointObject.py

```diff
@@ -0,0 +1,227 @@
+                "App::Property",
+                "This is the local coordinate system within the object2 that will be used to joint.",
+            ),
+        )
+
+        joint.Proxy = self
+        self.joint = joint
+
+    def onChanged(self, fp, prop):
+        """Do something when a property has changed"""
+        App.Console.PrintMessage("Change property: " + str(prop) + "\n")
+
+    def execute(self, fp):
+        """Do something when doing a recomputation, this method is mandatory"""
+        App.Console.PrintMessage("Recompute Python Box feature\n")
+
+    def setJointConnectors(self, current_selection):
+        # current selection is a vector of strings like "Assembly.Assembly1.Assembly2.Body.Pad.Edge16" including both what selection return as obj_name and obj_sub
+
+        # Start by clearing.
+        self.joint.Object1 = None
+        self.joint.Element1 = ""
+        self.joint.Vertex1 = ""
+        self.joint.Placement1 = UtilsAssembly.activeAssembly().Placement
+
+        self.joint.Object2 = None
+        self.joint.Element2 = ""
+        self.joint.Vertex2 = ""
+        self.joint.Placement2 = UtilsAssembly.activeAssembly().Placement
+
+        if len(current_selection) >= 1:
+            self.joint.Object1 = UtilsAssembly.getObject(current_selection[0])
+            self.joint.Element1 = UtilsAssembly.getElementName(current_selection[0])
+            self.joint.Vertex1 = ""
+            self.joint.Placement1 = findPlacement(
+                self.joint.Object1, self.joint.Element1, self.joint.Vertex1
+            )
+
+        if len(current_selection) >= 2:
+            self.joint.Object2 = UtilsAssembly.getObject(current_selection[1])
+            self.joint.Element2 = UtilsAssembly.getElementName(current_selection[1])
+            self.joint.Vertex2 = ""
+            self.joint.Placement2 = findPlacement(
+                self.joint.Object2, self.joint.Element2, self.joint.Vertex2
+            )
+
+    """
+    So here we want to find a placement that correspong to a local cordinate system that would be placed at the selected vertex.
+    - obj is usually a App::Link to a PartDesign::Body, or primitive, fasteners. But can also be directly the object.1
+    - elt can be a face, an edge or a vertex.
+    - If elt is a vertex, then vtx = elt And placement is vtx coordinates without rotation.
+    - if elt is an edge, then vtx = edge start/end vertex depending on which is closer. If elt is an arc or circle, vtx can also be the center. The rotation is the plane normal to the line positioned at vtx. Or for arcs/circle, the plane of the arc.
+    - if elt is a plane face, vtx is the face vertex (to the list of vertex we need to add arc/circle centers) the closer to the mouse. The placement is the plane rotation positioned at vtx
+    - if elt is a cylindrical face, vtx can also be the center of the arcs of the cylindrical face.
+    """
+
+    def findPlacement(obj, elt, vtx):
+        plc = obj.Placement
+
+    return plc
```

Comment:
```suggestion
        return plc
```
return outside of method, gives error when loading workbench:
![asm_error](https://github.com/FreeCAD/FreeCAD/assets/82042580/863472c4-e2ed-475a-afcf-ac589e0e630c)

Reply:
Sorry about that typo !